### PR TITLE
forge: support commit comments

### PR DESCRIPTION
**Pull request by edvbld**
Hi all,

please review this patch that adds the possibility to add a comment on a particular commit to `HostedRepository` via the method `addCommitComment`. To start out with I've only added support for adding a general comment about a whole commit, not a comment for a particular line in a commit. 

The comments for a commit can also be listed with the method `commitComments`.

Testing:
- Not that much, later patches will build upon the APIs introduced in this patch

Thanks,
Erik

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Reviewers
 * Robin Westberg ([rwestberg](@rwestberg) - **Reviewer**) ⚠️ Review applies to b5a30420b23c5afb010d165fa99cf417c2d15ab0

### Download
`$ git fetch https://git.openjdk.java.net/skara pull/639/head:pull/639`
`$ git checkout pull/639`

### CODE REVIEW

File: forge/src/main/java/org/openjdk/skara/forge/CommitComment.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
```
Comment (rwestberg) [inline forge/src/main/java/org/openjdk/skara/forge/CommitComment.java:2]: 2020!

File: forge/src/main/java/org/openjdk/skara/forge/CommitComment.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+     * Returns the relative path of the file.
+     */
+    public Optional<Path> path() {
+        return Optional.ofNullable(path);
+    }
+
+    /**
+     * Returns the line index in the diff.
+     */
+    public Optional<Integer> position() {
```
Comment (rwestberg) [inline forge/src/main/java/org/openjdk/skara/forge/CommitComment.java:65]: Perhaps drop this one? The other API's only expose a line number

File: forge/src/main/java/org/openjdk/skara/forge/CommitComment.java (hunk 1 above)
Comment (edvbld) [reply forge/src/main/java/org/openjdk/skara/forge/CommitComment.java:2]: Thanks, will fix!

File: forge/src/main/java/org/openjdk/skara/forge/CommitComment.java (hunk 2 above)
Comment (edvbld) [reply forge/src/main/java/org/openjdk/skara/forge/CommitComment.java:65]: Ok, yeah, sounds good :+1:

**Pull request closed without merging**